examples/block.rs

```diff
@@ -1,123 +1,248 @@
-use std::{error::Error, io, time::Duration};
+use std::{
+    error::Error,
+    io::{stdout, Stdout},
+    ops::ControlFlow,
+    time::Duration,
+};
 
 use crossterm::{
-    event::{self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode},
+    event::{self, Event, KeyCode},
     execute,
     terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
 };
-use ratatui::{prelude::*, widgets::*};
+use itertools::Itertools;
+use ratatui::{
+    prelude::*,
+    widgets::{
+        block::{Position, Title},
+        Block, BorderType, Borders, Padding, Paragraph, Wrap,
+    },
+};
 
-fn main() -> Result<(), Box<dyn Error>> {
-    // setup terminal
-    enable_raw_mode()?;
-    let mut stdout = io::stdout();
-    execute!(stdout, EnterAlternateScreen, EnableMouseCapture)?;
-    let backend = CrosstermBackend::new(stdout);
-    let mut terminal = Terminal::new(backend)?;
+type Frame<'a> = ratatui::Frame<'a, CrosstermBackend<Stdout>>;
+type Terminal = ratatui::Terminal<CrosstermBackend<Stdout>>;
+type Result<T> = std::result::Result<T, Box<dyn Error>>;
```

Comment:
Why exactly these type aliases are needed and simple imports are not sufficient?

(I can imagine a situation where a Rust beginner using these examples and asking the same question.)

Reply:
The idea behind this was to avoid repetition of the generic constraints in the render_* methods.
- The `Result` type alias is a fairly common practice to see, both in app code and library code (e.g. std::io::Result. In app code often people use `use anyhow::Result`).
- The `Frame` alias would expand to either `fn render_xxx<B: Backend>(frame: Frame<B>)` or the full `fn render_xxx(frame: Frame<'a, CrosstermBackend<Stdout>>)` across 11 methods that are arranged that way to make it easy to see structurally what's happening. 
- The `Terminal` alias is to my mind the most debatable choice as it's only 4 usages. It's there mainly for consistency with the choice to alias `Frame`.

I see these as simplifications generally, but given your question about why they are here, they sound like they could be changes that add complexity rather than reduce it. Would a comment explaining them work better? Perhaps:
```suggestion
// type aliases to reduce repetition of the generic parameters in method parameters
type Frame<'a> = ratatui::Frame<'a, CrosstermBackend<Stdout>>;
type Terminal = ratatui::Terminal<CrosstermBackend<Stdout>>;
type Result<T> = std::result::Result<T, Box<dyn Error>>;
```

Reply:
Yeah, I would say let's add a comment to make their usage clear-er.

Reply:
The newer version has the comments